File: lib/domain/crop/entities/crop_task_template_entity.rb
# frozen_string_literal: true

module Domain
  module Crop
    module Entities
      class CropTaskTemplateEntity
        attr_reader :id,
          :crop_id,
          :agricultural_task_id,
          :name,
          :description,
          :time_per_sqm,
          :weather_dependency,
          :required_tools,
          :skill_level,
          :created_at,
          :updated_at

        def initialize(
          id:,
          crop_id:,
          agricultural_task_id:,
          name:,
          description:,
          time_per_sqm:,
          weather_dependency:,
          required_tools:,
          skill_level:,
          created_at:,
          updated_at:
        )
          @id = id
          @crop_id = crop_id
          @agricultural_task_id = agricultural_task_id
          @name = name
          @description = description
          @time_per_sqm = time_per_sqm
          @weather_dependency = weather_dependency
          @required_tools = required_tools || []
          @skill_level = skill_level
          @created_at = created_at
          @updated_at = updated_at
        end
      end
    end
  end
end

File: crates/agrr-domain/src/crop/entities/crop_task_template_entity.rs
use rust_decimal::Decimal;

/// Ruby: `Domain::Crop::Entities::CropTaskTemplateEntity`
#[derive(Debug, Clone, PartialEq)]
pub struct CropTaskTemplateEntity {
    pub id: i64,
    pub crop_id: i64,
    pub agricultural_task_id: i64,
    pub name: String,
    pub description: Option<String>,
    pub time_per_sqm: Option<Decimal>,
    pub weather_dependency: Option<String>,
    pub required_tools: Vec<String>,
    pub skill_level: Option<String>,
    pub created_at: Option<String>,
    pub updated_at: Option<String>,
}
